### ISSUE TITLE
Add support for arguments in custom messages in `*-pattern` rules

### ISSUE THREAD

**Issue by jhae-de** [stylelint-scss/stylelint-scss#1186]
Currently, it is not possible to pass arguments to a custom message in the `*-pattern` rules.

Example:

```javascript
{
  'scss/at-function-pattern': [
    '^(-?[a-z][a-z0-9]*)(-[a-z0-9]+)*$',
    {
      message: (name, pattern) => `Expected "${name}" to match pattern "${pattern}"`,
    },
  ],
}
```

Results in:

```
Expected "undefined" to match pattern "undefined"
```

This applies to the following rules:
- `at-function-pattern`
- `at-mixin-pattern`
- `dollar-variable-pattern`
- `percent-placeholder-pattern`

I'm currently working on adding this feature to the rules in question.
